scripts/jenkins/jenkins-job-pipeline-report.py

```diff
@@ -0,0 +1,216 @@
+#!/usr/bin/python
+
+#
+# (c) Copyright 2018 SUSE LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+#
+# Generates a summary report based on the Jenkins pipeline API results.
+
+import argparse
+import json
+
+import jenkins
+import os
+import re
+import sys
+
+BUILD_INFO = jenkins.BUILD_INFO
+WORKFLOW_BUILD_INFO = '%(folder_url)sjob/%(short_name)s/%(number)d/'
+WORKFLOW_INFO = 'wfapi/describe'
+WORKFLOW_STAGE_INFO = 'execution/node/%(stage)s/wfapi/describe'
+WORKFLOW_STAGE_LOG = 'execution/node/%(stage)s/wfapi/log'
+PIPELINE_URL = 'blue/organizations/jenkins/%(name)s/detail/' \
+               '%(name)s/%(number)s/pipeline'
+PIPELINE_STAGE_URL = 'blue/organizations/jenkins/%(name)s/detail/' \
+                     '%(name)s/%(number)s/pipeline/%(stage)s'
+
+
+class Jenkins(jenkins.Jenkins):
```

Comment:
Any reason not to give this a class name indicating the fact that it is an enhanced super class of the standard Jenkins class? E.g. WorkflowJenkins?

Past experience suggests it can be helpful when tracing through the code to use unique names for super classes unless there is a strong reason to re-use the original name.

Reply:
Good point. Done.